### DISCUSSION TITLE
Using getLocalTimeZone() in RSC's?

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [adobe/react-spectrum#7042] [Q&A]
When you use methods like `parseAbsolute(), today(), isToday()` etc. they require you to provide an timeZone. The docs suggest `getLocalTimeZone()`.  

Using NextJs and creating RSC's you may sometime need use these methods and getLocalTimeZone uses `new Intl.DateTimeFormat().resolvedOptions().timeZone` under the hood. I would think this uses the server timeZone then.
What is the suggested way of providing the correct user  timeZone doing SSR? 
Maybe a stupid question, but I would to hear what you think. 😅

**Comment by reidbarber** [comment]
You should be able to set a cookie, then read it from the header in RSCs: https://nextjs.org/docs/app/api-reference/functions/cookies

**Reply by J4v4Scr1pt** [reply]
Interesting suggestion, I actully do that to provide the user locale to i18Next. 
In my case I'm using NextUI which is built upon react-spectrum. You need to have an NextUI wrapper to use theming, navigation etc. but you also provide an locale that is consumed by the react-aria I18nProvider under the hood. 

Long story short, could the I18nProvider's locale e.g en-US used somehow to determine the timeZone? or are they two completely seperate things?

**Reply by reidbarber** [reply]
They are separate things, but you can get a list of supported timezones by locale: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getTimeZones

I haven't tested this, so not sure how reliable it is.

**Reply by J4v4Scr1pt** [reply]
Ok thx again. I will play arround with this and test. But now I know a bit more how we should think.
